### PR TITLE
Add ability to specify pod_template_file in executor_config

### PR DESCRIPTION
**Pull request by dimberman**
Users will be able to override the base pod_template_file on a per-task
basis. This is valuable for users who want to store multiple base pod_template_files
and pick on a per-task basis!

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by kaxil** [comment]
I have rebased your PR on latest Master since we have applied [Black](https://github.com/apache/airflow/commit/4e8f9cc8d02b29c325b8a5a76b4837671bdf5f68) and [PyUpgrade](https://github.com/apache/airflow/commit/8c42cf1b00c90f0d7f11b8a3a455381de8e003c5) on Master.

File: airflow/executors/kubernetes_executor.py [hunk 1]
```diff
@@ -505,6 +506,20 @@ def create_pod_id(dag_id: str, task_id: str) -> str:
     return safe_dag_id + safe_task_id
 
 
+def get_base_pod_from_template(pod_template_file: Optional[str], kube_config: Any) -> k8s.V1Pod:
+    """
+    Reads either the pod_template_file set in the executor_config or the base pod_template_file
+    set in the airflow.cfg to craft a "base pod" that will be used by the KubernetesExecutor
+    @param pod_template_file: absolute path to a pod_template_file.yaml or None
+    @param kube_config:
+    @return:
```
Comment (kaxil) [inline airflow/executors/kubernetes_executor.py]: we use `:param` format

Comment (dimberman) [reply airflow/executors/kubernetes_executor.py]: @kaxil fixed

File: airflow/executors/kubernetes_executor.py [hunk 2]
```diff
@@ -505,6 +506,21 @@ def create_pod_id(dag_id: str, task_id: str) -> str:
     return safe_dag_id + safe_task_id
 
 
+def get_base_pod_from_template(pod_template_file: Optional[str], kube_config: Any) -> k8s.V1Pod:
+    """
+    Reads either the pod_template_file set in the executor_config or the base pod_template_file
+    set in the airflow.cfg to craft a "base pod" that will be used by the KubernetesExecutor
+
+    :param pod_template_file: absolute path to a pod_template_file.yaml or None
+    :param kube_config:
```
Comment (kaxil) [inline airflow/executors/kubernetes_executor.py]: Missing description

File: airflow/executors/kubernetes_executor.py [hunk 3]
```diff
@@ -505,6 +506,21 @@ def create_pod_id(dag_id: str, task_id: str) -> str:
...
+    :return:
```
Comment (kaxil) [inline airflow/executors/kubernetes_executor.py]: same here

**Review by kaxil — COMMENTED** [review]
Should the PR name be updated too? i.e. "Add ability to specify base Pod in executor_config"

**Comment by dimberman** [comment]
@kaxil changed

**Comment by kaxil** [comment]
Cool, can you address the remaining 2 comments please, once you do that, I am happy to approve this PR

**Comment by dimberman** [comment]
@dimberman adressed coments

**Comment by dimberman** [comment]
@kaxil addressed comments

**Review by kaxil — APPROVED** [review]

**Pull request merged**